### PR TITLE
Add models specs: join tables

### PR DESCRIPTION
**Pull request by ImMPrada**
This PR is part of Round 3 of the Milestone

```
ProfileLanguage
  associations
    is expected to belong to profile required: true
    is expected to belong to language required: true
  validations
    is expected to be valid
```

```
RepoLanguage
  associations
    is expected to belong to repo required: true
    is expected to belong to language required: true
  validations
    is expected to be valid
```

This PR is part of [PR 7](https://github.com/ImMPrada/github-profile-viewer-api/pull/7)

### CODE REVIEW

File: spec/models/profile_language_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  describe 'validations' do
+    subject(:profile_language) { create(:profile_language, profile: profile, language: language) }
+
+    let(:language) { create(:language) }
+    let(:location) { create(:location) }
+    let(:profile) { create(:profile, location: location) }
+
+    it { expect(profile_language).to be_valid }
+  end
```
Comment (mapra99) [inline spec/models/profile_language_spec.rb:17]: what are you trying to validate here? I don't see any validation on the model

File: spec/models/repo_language_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  describe 'validations' do
+    subject(:repo_language) { create(:repo_language, repo: repo, language: language) }
+
+    let(:language) { create(:language) }
+    let(:location) { create(:location) }
+    let(:profile) { create(:profile, location: location) }
+    let(:repo) { create(:repo, profile: profile) }
+
+    it { expect(repo_language).to be_valid }
+  end
```
Comment (mapra99) [inline spec/models/repo_language_spec.rb:18]: same here

File: spec/models/profile_language_spec.rb (hunk 1 above)
Comment (ImMPrada) [reply spec/models/profile_language_spec.rb]: I want to validate that the profile_language instance is created with provided profile and language.

Please take a look again

File: spec/models/repo_language_spec.rb (hunk 2 above)
Comment (ImMPrada) [reply spec/models/repo_language_spec.rb]: I want to validate that the repo_language instance is created with provided repo and language.

Please take a look again

File: spec/models/profile_language_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  describe 'validations' do
+    subject(:profile_language) { create(:profile_language, profile: profile, language: language) }
+
+    let(:language) { create(:language) }
+    let(:location) { create(:location) }
+    let(:profile) { create(:profile, location: location) }
+
+    it { expect(profile_language.profile).to eq(profile) }
+    it { expect(profile_language.language).to eq(language) }
```
Comment (mapra99) [inline spec/models/profile_language_spec.rb:17]: oh I see now. This test is exactly what shoulda_matchers is checking with the `describe` block you have above :)
```ruby
  describe 'associations' do
    it { is_expected.to belong_to(:profile) }
    it { is_expected.to belong_to(:language) }
  end
```
No need to do it yourself "manually"

File: spec/models/repo_language_spec.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+  describe 'validations' do
+    let(:repo_language) { create(:repo_language, repo: repo, language: language) }
+
+    let(:language) { create(:language) }
+    let(:repo) { create(:repo) }
+
+    it { expect(repo_language.repo).to eq(repo) }
+    it { expect(repo_language.language).to eq(language) }
```
Comment (mapra99) [inline spec/models/repo_language_spec.rb:16]: same as above

**Review by mapra99 — APPROVED** [review]
🚀

**Pull request merged**